listenbrainz/websockets/listens_dispatcher.py

```diff
@@ -1,81 +1,70 @@
 import json
-import pika
 import time
-import threading
+
+from kombu.mixins import ConsumerMixin
 
 from listenbrainz.utils import get_fallback_connection_name
 
-from flask import current_app
-from listenbrainz.webserver.views.api_tools import LISTEN_TYPE_PLAYING_NOW, LISTEN_TYPE_IMPORT
+from kombu import Connection, Exchange, Queue, Consumer
+from kombu.utils.debug import setup_logging
+
+setup_logging()
 
 
-class ListensDispatcher(threading.Thread):
+class ListensDispatcher(ConsumerMixin):
 
     def __init__(self, app, socketio):
-        threading.Thread.__init__(self)
         self.app = app
         self.socketio = socketio
+        self.connection = None
+        self.channel2 = None
 
-    def send_listens(self, listens, listen_type):
-        if listen_type == LISTEN_TYPE_PLAYING_NOW:
-            event_name = 'playing_now'
-        else:
-            event_name = 'listen'
-        for listen in listens:
-            self.socketio.emit(event_name, json.dumps(listen), to=listen['user_name'])
-
-    def callback_listen(self, channel, method, properties, body):
-        listens = json.loads(body)
-        self.send_listens(listens, LISTEN_TYPE_IMPORT)
-        channel.basic_ack(delivery_tag=method.delivery_tag)
+        self.unique_exchange = Exchange(app.config["UNIQUE_EXCHANGE"], "fanout", durable=False)
```

Comment:
We should probably be using durable=True exchanges. However, the existing ones are not durable so we cannot switch deleting the existing ones. Something to do later.